pkg/secrethub/credentials/providers.go

```diff
@@ -39,7 +39,12 @@ func (k KeyProvider) Passphrase(passphraseReader Reader) Provider {
 // Provide implements the Provider interface for a KeyProvider.
 func (k KeyProvider) Provide(httpClient *http.Client) (auth.Authenticator, Decrypter, error) {
 	key, err := ImportKey(k.credentialReader, k.passphraseReader)
-	if err != nil {
+	if source, ok := k.credentialReader.(CredentialSource); ok && err != nil {
+		return nil, nil, ErrLoadingCredential{
+			Location: source.Source(),
+			Err:      err,
+		}
+	} else if err != nil {
```

Comment:
Interestingly, errors regarding the header of the credentials are not wrapped properly, but ones regarding the payload are.

Reply:
Yes! And the `ImportKey` function is wrapped everywhere except the `account init` command and the `config update passphrase command`. I think it might be a good idea to also wrap the error there